apps/zero-runtime-next-app/src/components/Avatar/Avatar.js

```diff
@@ -0,0 +1,271 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_composeClasses as composeClasses } from '@mui/base/composeClasses';
+import { styled } from '@mui/zero-runtime';
+import { useThemeProps } from '@mui/material/styles';
+import Person from '@mui/material/internal/svg-icons/Person';
+import { getAvatarUtilityClass } from '@mui/material/Avatar';
+
+const useUtilityClasses = (ownerState) => {
+  const { classes, variant, colorDefault } = ownerState;
+
+  const slots = {
+    root: ['root', variant, colorDefault && 'colorDefault'],
+    img: ['img'],
+    fallback: ['fallback'],
+  };
+
+  return composeClasses(slots, getAvatarUtilityClass, classes);
+};
+
+const AvatarRoot = styled('div', {
+  name: 'MuiAvatar',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[ownerState.variant],
+      ownerState.colorDefault && styles.colorDefault,
+    ];
+  },
+})(({ theme }) => ({
+  position: 'relative',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  flexShrink: 0,
+  width: 40,
+  height: 40,
+  fontFamily: theme.typography.fontFamily,
+  fontSize: theme.typography.pxToRem(20),
+  lineHeight: 1,
+  borderRadius: '50%',
+  overflow: 'hidden',
+  userSelect: 'none',
+  variants: [
+    {
+      props: { variant: 'rounded' },
+      style: {
+        borderRadius: (theme.vars || theme).shape.borderRadius,
+      },
+    },
+    {
+      props: { variant: 'square' },
+      style: {
+        borderRadius: 0,
+      },
+    },
+    {
+      props: { colorDefault: true },
+      style: {
+        color: (theme.vars || theme).palette.background.default,
+        ...(theme.vars
+          ? {
+              backgroundColor: theme.vars.palette.Avatar.defaultBg,
+            }
+          : {
+              backgroundColor:
+                theme.palette.mode === 'light' ? theme.palette.grey[400] : theme.palette.grey[600],
```

Comment:
For light/dark mode, this won't work. We'll have to use `theme.applyDarkStyles()` and similarly to rtl/ltr, we'll need another utility.